src/component/Legend.tsx

```diff
@@ -103,14 +101,37 @@ export class Legend extends PureComponent<Props, State> {
     return null;
   }
 
-  private getBBoxSnapshot() {
-    const { boxWidth, boxHeight } = this.state;
+  private updateBBox() {
+    const { onBBoxUpdate } = this.props;
+
+    if (this.wrapperNode && this.wrapperNode.getBoundingClientRect) {
+      const box = this.wrapperNode.getBoundingClientRect();
 
-    if (boxWidth >= 0 && boxHeight >= 0) {
-      return { width: boxWidth, height: boxHeight };
+      if (
+        Math.abs(box.width - this.lastBoundingBox.width) > EPS ||
+        Math.abs(box.height - this.lastBoundingBox.height) > EPS
+      ) {
```

Comment:
It would be great to extract more named helper function to make the code more readable. 

```
areBoxesDifferent(oneBox, anotherBox){
   return  Math.abs(box.width - this.lastBoundingBox.width) > EPS ||
        Math.abs(box.height - this.lastBoundingBox.height) > EPS
}
```